### ISSUE TITLE
Support other maze generation algos?

### ISSUE THREAD

**Issue by phayes** [kbsymanz/maze-rust#2]
It would be wonderful to support all sorts of maze generation algorithms such as 

 - Depth First Search
 - Kruskal's Algorithm
 - Wilson's Algorithm
 - Eller's Algorithm
 - Prim's Algorithm

**Comment by kbsymanz** [comment]
Agreed, that would be a good idea. I haven't worked in Rust in a while and this repo is out of date. But I would gladly entertain a pull requests that add more algorithms while bringing the dependencies to current. Or feel free to fork yourself if you prefer.
